src/main/java/org/springframework/data/redis/connection/lettuce/LettuceConverters.java

```diff
@@ -199,6 +208,74 @@ public Tuple convert(ScoredValue<byte[]> source) {
 			}
 		};
 
+		PARTITIONS_TO_CLUSTER_NODES = new Converter<Partitions, List<RedisClusterNode>>() {
+
+			@Override
+			public List<RedisClusterNode> convert(Partitions source) {
+
+				if (source == null) {
+					return Collections.emptyList();
+				}
+				List<RedisClusterNode> nodes = new ArrayList<RedisClusterNode>();
+				for (com.lambdaworks.redis.cluster.models.partitions.RedisClusterNode node : source.getPartitions()) {
+					nodes.add(CLUSTER_NODE_TO_CLUSTER_NODE_CONVERTER.convert(node));
+				}
+
+				return nodes;
+			};
+
+		};
+
+		CLUSTER_NODE_TO_CLUSTER_NODE_CONVERTER = new Converter<com.lambdaworks.redis.cluster.models.partitions.RedisClusterNode, RedisClusterNode>() {
+
+			@Override
+			public RedisClusterNode convert(com.lambdaworks.redis.cluster.models.partitions.RedisClusterNode source) {
+
+				Set<Flag> flags = parseFlags(source.getFlags());
+
+				return RedisClusterNode.newRedisClusterNode().listeningAt(source.getUri().getHost(), source.getUri().getPort())
```

Comment:
Lettuce returns `null` for RedisURI if the cluster node has no address (`noaddr` flag, see https://github.com/mp911de/lettuce/issues/173)